### PR TITLE
Initial openvpn-client role

### PR DESCRIPTION
**Pull request by Scriptkiddi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+| Variable                          | Default / Mandatory        | Description                                                                                |
+|-----------------------------------|----------------------------|--------------------------------------------------------------------------------------------|
+| `openvpn_client_server`           | :heavy_multiplication_x:   | Inventory name of the host running the [openvpn role](https://github.com/stuvusIT/openvpn) |
+| `openvpn_client_name`             | `{{ inventory_hostname }}` | Client name used in on the openvpn server                                                  |
+| `openvpn_client_config_file_name` | `openvpn.ovpn`             | File name to copy over, only used when `openvpn_client_server` is not set                  |
+
+## Example Playbook
+
+Including an example of how to use your role (for instance, with variables passed in as parameters) is always nice for users too:
```
Comment (SlothOfAnarchy) [inline README.md]: Indeed it is.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Requirements
+
+An openvpn config file either on the openvpn server or in the inventory and an apt based system.
+
+## Role Variables
+
+| Variable                          | Default / Mandatory        | Description                                                                                |
+|-----------------------------------|----------------------------|--------------------------------------------------------------------------------------------|
+| `openvpn_client_server`           | :heavy_multiplication_x:   | Inventory name of the host running the [openvpn role](https://github.com/stuvusIT/openvpn) |
```
Comment (SlothOfAnarchy) [inline README.md]: Please state how the config file on the server has to be named and where this role will search for it.

File: tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  notify: Restart openvpn service
+
+- name: Copy to host
+  copy:
+    src: "{{ openvpn_client_config_file_name }}"
+    dest: "/etc/openvpn/{{ openvpn_client_name }}.conf"
+  when: openvpn_client_server is not defined
+  notify: Restart openvpn service
+
+
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: Unnecessary newline

File: tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+- name: Copy to host
+  copy:
+    content: "{{ openvpn_config_file['content'] | b64decode }}"
+    dest: "/etc/openvpn/{{ openvpn_client_name }}-{{ openvpn_client_server }}.conf"
+  when: openvpn_client_server is defined
+  notify: Restart openvpn service
+
+- name: Copy to host
+  copy:
+    src: "{{ openvpn_client_config_file_name }}"
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: In the past we used a structure in `files` that takes files from a subfolder named after the host.
Does this work for this case, too or would the path have to be specified in this role?

File: tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    name: openvpn
+
+- name: Slurp openvpn config
+  slurp:
+    src: "{{ hostvars[openvpn_client_server].openvpn_base_dir }}/{{ openvpn_client_name }}-{{ openvpn_client_server }}.ovpn"
+  register: openvpn_config_file
+  delegate_to: "{{ openvpn_client_server }}"
+  when: openvpn_client_server is defined
+
+- name: Copy to host
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: Copy OpenVPN config to host

same below

File: defaults/main.yml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+openvpn_client_config_file_name: openvpn.ovpn
+openvpn_client_name: "{{ inventory_hostname }}"
+
```
Comment (SlothOfAnarchy) [inline defaults/main.yml]: why the newline?

File: defaults/main.yml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+openvpn_client_config_file_name: openvpn.ovpn
```
Comment (SlothOfAnarchy) [inline defaults/main.yml:2]: `---`

File: README.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
+# openvpn-client
+
+This role installs and copys an openvpn configuration, either from the [openvpn server](https://github.com/stuvusIT/openvpn) or from files in the inventory
```
Comment (SlothOfAnarchy) [inline README.md]: `.` after a full sentence ;)

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]
mostly minor things, documentation and questions

File: README.md (hunk 1 above)
Comment (Scriptkiddi) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (Scriptkiddi) [reply README.md]: no i clearly state that you have to use my role to do this so this is irrelevant

File: tasks/main.yml (hunk 3 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: tasks/main.yml (hunk 4 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: tasks/main.yml (hunk 5 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: fixed

File: defaults/main.yml (hunk 7 above)
Comment (Scriptkiddi) [reply defaults/main.yml:2]: fixed

**Comment by Scriptkiddi** [comment]
@SlothOfAnarchy pls rereview

**Review by SlothOfAnarchy — APPROVED** [review]

**Pull request merged**